### PR TITLE
Update: Implement auto fix for `indent` rule (fixes #3734)

### LINKED ISSUE #3734: Implement auto fix functionality for indent rule
Implement auto fix functionality for indent rule

### PR DESCRIPTION
**Pull request by gyandeeps**

### CODE REVIEW

File: lib/rules/indent.js [hunk 1]
```diff
@@ -83,20 +83,78 @@ module.exports = function(context) {
...
             gotten: gotten
         };
+        var indentChar = indentType === "space" ? " " : "\t";
+
+        /**
+         * Responsible for fixing the indentation issue fix
+         * @returns {Function} function to be executed by the fixer
+         * @private
+         */
+        function fixerFunction() {
```
Comment (platinumazure) [inline lib/rules/indent.js]: No verb here. Maybe `getFixerFunction` or just `getFix`?

Comment (nzakas) [inline lib/rules/indent.js]: Agree

**Comment by gyandeeps** [comment]
Changes have been made.

File: tests/lib/rules/indent.js [hunk 2]
```diff
@@ -838,7 +839,12 @@ ruleTester.run("indent", rule, {
                 "b();\n" +
                 "}\n",
             options: [2],
-            errors: expectedErrors([[3, 2, 0, "ExpressionStatement"]])
+            errors: expectedErrors([[3, 2, 0, "ExpressionStatement"]]),
```
Comment (nzakas) [inline tests/lib/rules/indent.js]: Can you put `output` next to `code` in the tests? Makes it easier to see issues.

Comment (gyandeeps) [inline tests/lib/rules/indent.js]: Change have been made.

**Comment by nzakas** [comment]
Looks great! Very excited about this one.

**Comment by gyandeeps** [comment]
me 2... :tada:

**Pull request merged**